src/Lykke.Job.BlockchainCashinDetector/Modules/CqrsModule.cs

```diff
@@ -74,38 +100,37 @@ protected override void Load(ContainerBuilder builder)
             // Projections
             builder.RegisterType<ClientOperationsProjection>();
             builder.RegisterType<MatchingEngineCallDeduplicationsProjection>();
+        }
 
-            builder.Register(ctx => CreateEngine(ctx, messagingEngine))
-                .As<ICqrsEngine>()
-                .SingleInstance()
-                .AutoActivate();
+        protected virtual IEndpointResolver GetDefaultEndpointResolver()
+        {
+            return new RabbitMqConventionEndpointResolver(
+                "RabbitMq",
+                SerializationFormat.MessagePack,
+                environment: "lykke");
         }
 
-        private CqrsEngine CreateEngine(IComponentContext ctx, IMessagingEngine messagingEngine)
+        protected CqrsEngine CreateEngine(IComponentContext ctx)
         {
+            var logFactory = ctx.Resolve<ILogFactory>();
+            var messagingEngine = RegisterMessagingEngine(ctx);
             var defaultRetryDelay = (long)_settings.RetryDelay.TotalMilliseconds;
 
             const string defaultPipeline = "commands";
             const string defaultRoute = "self";
             const string eventsRoute = "evets";
 
-            return new CqrsEngine(
-                _log,
-                ctx.Resolve<IDependencyResolver>(),
-                messagingEngine,
-                new DefaultEndpointProvider(),
-                true,
-                Register.DefaultEndpointResolver(new RabbitMqConventionEndpointResolver(
-                    "RabbitMq",
-                    SerializationFormat.MessagePack,
-                    environment: "lykke")),
+#pragma warning disable CS0612 // Type or member is obsolete
 
+            var registration = new List<IRegistration>()
+            {
+                Register.DefaultEndpointResolver(GetDefaultEndpointResolver()),
                 Register.BoundedContext(Self)
                     .FailedCommandRetryDelay(defaultRetryDelay)
 
                     .ListeningCommands(typeof(LockDepositWalletCommand))
                     .On(defaultRoute)
-                    .WithLoopback()
+                    .WithLoopback()//When it is sent not from saga
```

Comment:
Please, double check: after my last conversation with Mihail Musin, I'm not sure, that .WithLoopback is necessary

Reply:
That code was before my changes =)

Reply:
It works correctly

Reply:
+